Euri-backend/Repository/BasketRepository.cs

```diff
@@ -0,0 +1,88 @@
+using Euri_backend.Data;
+using Euri_backend.Data.Models;
+using Euri_backend.Repository.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace Euri_backend.Repository;
+
+public class BasketRepository: IBasketRepository
+{
+    private readonly AppDbContext _ctxt;
+    
+    public BasketRepository(AppDbContext ctxt)
+    {
+        _ctxt = ctxt;
+    }
+    
+    public async Task<BasketModel> GetBasket(int userId)
+    {
+        return await _ctxt.Baskets
+            .Include(x => x.Items)
+            .ThenInclude(items => items.Product)
+            .FirstOrDefaultAsync(x => x.Id == userId);
+    }
+
+    public async Task<IEnumerable<BasketModel>> GetAllBaskets()
+    {
+        return await _ctxt.Baskets
+            .Include(x => x.Items)
+            .ThenInclude(items => items.Product)
+            .ToListAsync();
+    }
+
+    public async Task<BasketModel> CreateBasket(BasketModel basket)
+    {
+         _ctxt.Baskets.Add(basket);
+        await _ctxt.SaveChangesAsync();
+
+        var basketEntity = await GetBasket(basket.Id);
+        return basketEntity;
+    }
+
+    public async Task<BasketModel> UpdateBasket(BasketModel basket)
+    {
+        try
+        {
+            var basketEntity = await GetBasket(basket.Id);
+            basketEntity.Items.Clear();
+            
+            basketEntity.Items = basket.Items;
```

Comment:
Dit zou je in feite niet mogen doen. Het is beter dat collecties zelfs geen setter hebben. Er kunnen namelijk event handlers aan gekoppeld zijn die dan niet meer gaan werken en zelfs voor memory leaks gaan zorgen. 
Als alternatief kan je de AddRange methode gebruiken nadat je de Clear hebt gedaan.